BridgeSDKTests/SBBParticipantManagerIntegrationTests.m

```diff
@@ -284,10 +287,50 @@ - (void)testSetSharingScope
 
 - (void)testSetExternalIdentifier
 {
+    NSMutableDictionary *headers = [NSMutableDictionary dictionary];
+    [gAdminAuthManager addAuthHeaderToHeaders:headers];
+
+    // Create an external identifier to test with
+    NSString *externalIdentifierFormat = @"test-external-identifier%@";
+    NSString *substudyId = @"test-external-id-substudy";
+    NSString *unique = [[NSProcessInfo processInfo] globallyUniqueString];
+    __block NSString *externalIdentifier = [NSString stringWithFormat:externalIdentifierFormat, unique];
+    NSDictionary *externalIdObject =
+    @{
+      @"identifier": externalIdentifier,
+      @"substudyId": substudyId,
+      @"type": @"ExternalIdentifier"
+      };
+    
+    XCTestExpectation *expectCreatedIdentifier = [self expectationWithDescription:@"created external identifier"];
+    id<SBBBridgeNetworkManagerProtocol> bridgeMan = SBBComponent(SBBBridgeNetworkManager);
+    [bridgeMan post:DEV_EXTERNAL_ID_API headers:headers parameters:externalIdObject completion:^(NSURLSessionTask *task, id responseObject, NSError *error) {
+        if (error) {
+            NSLog(@"Failed to create external ID %@\nError:%@\nResponse:%@", externalIdentifier, error, responseObject);
+            externalIdentifier = nil;
+        } else {
+            NSLog(@"Created test external ID %@", externalIdentifier);
+        }
+        
+        [expectCreatedIdentifier fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:5.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"Timeout creating external identifier %@: %@", externalIdentifier, error);
+            externalIdentifier = nil;
+        }
+    }];
+
+    if (!externalIdentifier) {
+        // If we couldn't create the externalIdentifier, just fail & bail (we've already logged it)
+        XCTAssert(NO);
```

Comment:
Does Obj-c have `XCTFail(@"Impossible code path");` ? It's always nice to be able to look at the travis logs and see a message for why it failed. Travis doesn't tell you *where* it failed.